site/_posts/2021-03-23-migration-types.md

```diff
@@ -0,0 +1,107 @@
+
+In general, the segment by segment migration is useful when you have natural, well, segments. It decreases cutover risk because a smaller chunk of users will be migrated, and you can control which ones you do first. But it also introduces more complexities because you are running multiple migrations.
+
+## Slow Migration
+
+This approach is the logical endpoint of segment by segment migration. What if you made each chunk of accounts to be migrated as small as possible. Well, you end up with each segment being a single user. 
+
+With the slow migration approach, you need to: 
+
+* Map your user attributes from the old system to the new system. 
+* Set up a connection between the legacy system and FusionAuth. This could be an API call or some other way to pass authentication information from FusionAuth to the old system, and get back an authentication response.
+* Modify the application or applications to send users to FusionAuth for authentication. Test these changes before rolling them out, including that the connection created above works. If you previously used OIDC or SAML to connect to the legacy system, this effort may be minimal.
+* At this point, FusionAuth receives all auth requests. It delegates the initial authentication for each user to the legacy system. 
+* When the first call happens, the old system returns account data if the user presented valid credentials. FusionAuth creates a new user with the data. 
+* When this user logs in subsequently, FusionAuth responds; it is now the system of record, the user has been migrated, and the old system no longer has that particular user's information.
+
+This approach has a number of benefits. You are only migrating at the time a user authenticates. Therefore if there are issues, the blast radius is smaller and limited to the user authenticating. You have the user's password (they're providing it when they are authenticating), so you can upgrade their password hash to use a different algorithm transparently. This approach sidesteps any complexities around porting over bespoke hashing algorithms as well. 
+
+Application cutover is, all other things equal, simpler. You aren't moving any account data, which can often take some time. Instead you are only switching where users authenticate. In addition, you can scrub your user base of inactive users with little effort. Or, before the migration, you can contact them and encourage them to log in to your applications, possibly regaining their interest.
+
+However, a slow migration isn't always the best solution. Some things to consider include:
+
+* You are passing a plaintext password or other credential from FusionAuth to the legacy system. Make sure you take care of this precious cargo using TLS at a minimum. If possible, keep it from travelling over the internet by placing FusionAuth and the legacy system on the same network.
```

Comment:
...cargo using TLS (or some other packet layer security protocol) ....(might not be a thing) 😄